internal/logger/logger.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the License. A copy of the License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+
+package logger
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/aws/aws-xray-sdk-go/xraylog"
```

Comment:
I would suggest you keep all logger related file into one folder instead of split into two folders.

Reply:
I split it in two so we can have the public API for implementing the logger, and the private API for writing to the logger.